src/main/java/dev/mccue/resolve/core/DependencyGraph.java

```diff
@@ -0,0 +1,146 @@
+package dev.mccue.resolve.core;
+
+import java.util.List;
+import java.util.Map;
+import java.util.HashMap;
+import java.util.ArrayList;
+
+import static dev.mccue.resolve.core.DependencyGraph.Interdependency.*;
+
+public class DependencyGraph {
+    private final Map<DependencyVertex, List<DependencyVertex>> deps = new HashMap<>();
+
+     public static enum Interdependency {
+        TwoSubOfOne,
+        OneSubOfTwo,
+        NoRelation
+    }
```

Comment:
Naming on the enum here is a tad weird but not in a way I care that much about for a class like this which probably will stay internal. *generally* its `TWO_SUB_OF_ONE` and such but thats not a law i suppose.